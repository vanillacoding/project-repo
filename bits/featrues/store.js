import { configureStore } from '@reduxjs/toolkit';
import ReduxThunk from 'redux-thunk';
import { createLogger } from 'redux-logger';
import { userSlice } from './userSlice';
import { habitSlice } from './habitSlice';
import AsyncStorage from '@react-native-async-storage/async-storage';
import { persistStore, persistReducer } from 'redux-persist';

const userPersistConfig = {
  key: 'user',
  storage: AsyncStorage,
  blacklist: ['isSuccess', 'isError', 'isFetching']
};

const habitPersistConfig = {
  key: 'habit',
  storage: AsyncStorage,
  blacklist: ['isSuccess', 'isError', 'isFetching']
};

const reducer = {
  user: persistReducer(userPersistConfig, userSlice.reducer),
  habit: persistReducer(habitPersistConfig, habitSlice.reducer)
};

const middleware = [ReduxThunk, createLogger()];

const store = configureStore({
  reducer,
  middleware
});

const persistedStore = persistStore(store);

export { store, persistedStore };
