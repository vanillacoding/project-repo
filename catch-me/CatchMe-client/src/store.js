import { configureStore } from "@reduxjs/toolkit";
import logger from "redux-logger"

import single from "./redux/slices/singlePlaySlice"; 
import multiple from "./redux/slices/multiplaySlice";
import socketMiddleware from "./redux/middleware/socket";

const store = configureStore({
  reducer: {
    single,
    multiple,
  },
  middleware: (getDefaultMiddleware) => {
    if (process.env.NODE_ENV !== "production") {
      return getDefaultMiddleware().concat([logger, socketMiddleware()]);
    }

    return getDefaultMiddleware().concat([logger, socketMiddleware()]);
  },
  devTools: process.env.NODE_ENV !== "production",
});

export default store;
