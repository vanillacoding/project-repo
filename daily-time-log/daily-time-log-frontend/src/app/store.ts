import { configureStore } from "@reduxjs/toolkit";
import { TypedUseSelectorHook, useDispatch, useSelector } from "react-redux";
import logger from "redux-logger";

import authReducer from "../features/auth/authSlice";
import calendarReducer from "../features/calendar/calendarSlice";
import schedulesReducer from "../features/schedules/schedulesSlice";
import milestonesReducer from "../features/milestones/milestonesSlice";
import timeLogReducer from "../features/timeLog/timeLogSlice";
import settingReducer from "../features/setting/settingSlice";

const store = configureStore({
  reducer: {
    auth: authReducer,
    calendar: calendarReducer,
    schedules: schedulesReducer,
    milestones: milestonesReducer,
    timeLog: timeLogReducer,
    setting: settingReducer,
  },
  middleware: (getDefaultMiddleware) => {
    return process.env.REACT_APP_NODE_ENV !== "production"
      ? getDefaultMiddleware().concat(logger)
      : getDefaultMiddleware();
  },
  devTools: process.env.REACT_APP_NODE_ENV !== "production",
});

export type RootState = ReturnType<typeof store.getState>;
export type AppDispatch = typeof store.dispatch;

export const useAppDispatch = () => useDispatch<AppDispatch>();
export const useAppSelector: TypedUseSelectorHook<RootState> = useSelector;

export default store;
