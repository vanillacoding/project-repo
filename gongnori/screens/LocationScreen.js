import React, { useState } from "react";
import { StyleSheet, View, Text } from "react-native";
import { useDispatch, useSelector } from "react-redux";
import { SafeAreaView } from "react-native-safe-area-context";

import produce from "immer";
import _ from "lodash";

import CustomButton from "../components/CustomButton";
import DropDown from "../components/DropDown";
import { saveMyLocation } from "../actions/userActionCreators";

import * as colors from "../constants/colors";
import * as fonts from "../constants/fonts";
import * as sizes from "../constants/sizes";
import * as params from "../constants/params";

export default function LocationScreen() {
  const [myLocations, setMyLocations] = useState([]);

  const locations = useSelector((state) => {
    return state.app.locations;
  }, (prev, next) => _.cloneDeep(prev) === _.cloneDeep(next));

  const dispatch = useDispatch();

  const locationOptions = locations.map((location) => {
    const { city, district } = location;

    return `${city} ${district}`;
  });

  const handleSelectLocation = (index) => {
    const isAlreadySelected = myLocations.some((location) => {
      return location.id === locations[index].id;
    });

    if (isAlreadySelected) { return }

    setMyLocations(produce(myLocations, (draft) => {
      draft.unshift(locations[index]);
      draft.splice(2);
    }));
  };

  const handlePressButton = () => _.throttle(() => {
    dispatch(saveMyLocation(myLocations));
  }, params.THROTTLE_TIME);

  return (
    <SafeAreaView style={styles.container}>
      <Text style={styles.title}>
        동네 두개를 골라주세요
      </Text>
      <View style={styles.dropDownContainer}>
        <DropDown
          defaultValue={"동네 1"}
          options={locationOptions}
          style={styles.field}
          onSelect={handleSelectLocation}
        />
      </View>
      <View style={styles.dropDownContainer}>
        <DropDown
          defaultValue={"동네 2"}
          options={locationOptions}
          style={styles.field}
          onSelect={handleSelectLocation}
        />
      </View>
      <View style={styles.buttonContainer}>
        <CustomButton
          title={"등록하기"}
          style={styles.button}
          onPress={handlePressButton}
        />
      </View>
    </SafeAreaView>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    alignItems: "center",
    backgroundColor: colors.SECONDARY_GRAY,
  },
  title: {
    width: 0.8 * sizes.DEVICE_WIDTH,
    height: 0.15 * sizes.DEVICE_HEIGHT,
    marginTop: 100,
    textAlign: "center",
    textAlignVertical: "center",
    fontSize: sizes.PRIMARY_FONT_SIZE,
    fontFamily: fonts.SECONDARY_FONT,
  },
  dropDownContainer: {
    justifyContent: "center",
    height: 0.1 * sizes.DEVICE_HEIGHT,
  },
  buttonContainer: {
    justifyContent: "center",
    height: 0.1 * sizes.DEVICE_HEIGHT,
  },
  field: {
    width: 0.5 * sizes.DEVICE_WIDTH,
    height: 0.06 * sizes.DEVICE_HEIGHT,
    borderRadius: 5,
    backgroundColor: colors.SECONDARY_WHITE,
    fontSize: sizes.QUATERNARY_FONT_SIZE,
    fontFamily: fonts.NOTO_SANS_KR_300_LIGHT,
    textAlignVertical: "center",
    textAlign: "center",
    includeFontPadding: false,
  },
  button: {
    width: 0.5 * sizes.DEVICE_WIDTH,
    height: 0.06 * sizes.DEVICE_HEIGHT,
    borderRadius: 10,
    backgroundColor: colors.PRIMARY_BLUE,
    color: colors.SECONDARY_WHITE,
    fontSize: sizes.SECONDARY_FONT_SIZE,
  },
});
