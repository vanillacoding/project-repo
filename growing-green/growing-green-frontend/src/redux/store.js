import { configureStore } from '@reduxjs/toolkit';
import thunk from 'redux-thunk';
import logger from 'redux-logger';
import rootReducer from './root-reducer';

let middleware = [thunk];

if (process.env.NODE_ENV === 'development') {
  middleware = [...middleware, logger];
}

const store = configureStore({
  reducer: rootReducer,
  middleware: middleware,
});

export default store;
