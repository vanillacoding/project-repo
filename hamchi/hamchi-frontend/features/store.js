import { configureStore } from '@reduxjs/toolkit';
import ReduxThunk from 'redux-thunk';
import { userSlice } from './userSlice';
import { postSlice } from './postSlice';
import { chatSlice } from './chatSlice';
import { filteredPostSlice } from './filteredPostSlice';
import { createLogger } from 'redux-logger';

const middleware = [ReduxThunk];

const store = configureStore({
  reducer:
  {
    user: userSlice.reducer,
    post: postSlice.reducer,
    chat: chatSlice.reducer,
    filteredPost: filteredPostSlice.reducer
  },
  middleware: middleware
});

export default store;
