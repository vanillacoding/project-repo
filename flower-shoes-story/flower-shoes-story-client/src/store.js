import { combineReducers } from "redux";
import { persistReducer } from "redux-persist";
import { configureStore } from "@reduxjs/toolkit";
import logger from "redux-logger";
import storage from "redux-persist/lib/storage";

import userReducer from "./features/userSlice";
import preloadReducer from "./features/preloadSlice";

const reducers = combineReducers({
  user: userReducer,
  preload: preloadReducer,
});

const persistConfig = {
  key: "root",
  storage,
  whitelist: ["user"],
};

const persistedReducer = persistReducer(persistConfig, reducers);

const store = configureStore({
  reducer: persistedReducer,
  devTools: process.env.NODE_ENV !== "production",
  middleware: [logger],
});

export default store;
