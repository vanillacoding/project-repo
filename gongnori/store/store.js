import { createStore, applyMiddleware } from "redux";
import thunk from "redux-thunk";
import { createLogger } from "redux-logger";
import { NODE_ENV } from "@env";
import reducer from "../reducers/reducer";

const middleware = [thunk];

if (NODE_ENV === "development") {
  middleware.push(createLogger());
}

export default store = createStore(reducer, applyMiddleware(...middleware));
