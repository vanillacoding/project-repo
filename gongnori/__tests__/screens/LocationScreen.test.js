import React from "react";
import { Provider } from "react-redux";
import renderer from "react-test-renderer";
import { render } from "@testing-library/react-native";
import LocationScreen from "../../screens/LocationScreen";
import store from "../../store/store";

describe("<LocationScreen />", () => {
  test("snapshot test", () => {
    const component = renderer.create(
      <Provider store={store}>
        <LocationScreen />
      </Provider>
    );

    expect(component).toMatchSnapshot();
  });

  test("indicator text should be rendered", () => {
    const { queryByText } = render(
      <Provider store={store}>
        <LocationScreen />
      </Provider>
    );

    expect(queryByText("동네 두개를 골라주세요")).not.toBeNull();
    expect(queryByText("동네 1")).not.toBeNull();
    expect(queryByText("동네 2")).not.toBeNull();
    expect(queryByText("등록하기")).not.toBeNull();
  });
});
