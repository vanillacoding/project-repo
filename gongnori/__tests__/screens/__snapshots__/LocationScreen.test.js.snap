// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LocationScreen /> snapshot test 1`] = `
<RNCSafeAreaView
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#EBEFF5",
      "flex": 1,
    }
  }
>
  <Text
    style={
      Object {
        "fontFamily": undefined,
        "fontSize": 24,
        "height": 200.1,
        "marginTop": 100,
        "textAlign": "center",
        "textAlignVertical": "center",
        "width": 600,
      }
    }
  >
    동네 두개를 골라주세요
  </Text>
  <View
    style={
      Object {
        "height": 133.4,
        "justifyContent": "center",
      }
    }
  >
    <View
      adjustFrame={[Function]}
      animated={true}
      defaultIndex={-1}
      defaultValue="동네 1"
      disabled={false}
      dropdownStyle={
        Object {
          "height": 0,
          "width": 375,
        }
      }
      dropdownTextStyle={
        Object {
          "color": "black",
          "fontFamily": "NotoSansKR_300Light",
          "fontSize": 14,
          "height": 24,
          "includeFontPadding": false,
          "marginVertical": 5,
          "paddingVertical": 0,
          "textAlign": "center",
          "textAlignVertical": "center",
        }
      }
      isFullWidth={false}
      keyboardShouldPersistTaps="never"
      multipleSelect={false}
      onSelect={[Function]}
      options={Array []}
      renderButtonComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "render": [Function],
        }
      }
      renderRightComponent={[Function]}
      renderRowComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "render": [Function],
        }
      }
      scrollEnabled={true}
      showsVerticalScrollIndicator={true}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#FFFFFF",
          "borderRadius": 5,
          "height": 80.03999999999999,
          "justifyContent": "center",
          "width": 375,
        }
      }
      testID="modal-drop-down"
      textStyle={
        Object {
          "color": undefined,
          "fontFamily": "NotoSansKR_400Regular",
          "fontSize": 14,
          "height": 80.03999999999999,
          "includeFontPadding": false,
          "textAlign": "center",
          "textAlignVertical": "center",
          "width": 375,
        }
      }
    >
      <View
        accessible={true}
        focusable={true}
        forwardedRef={[Function]}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
            }
          }
        >
          <Text
            numberOfLines={1}
            style={
              Array [
                Object {
                  "fontSize": 12,
                },
                Array [
                  Object {
                    "color": undefined,
                    "fontFamily": "NotoSansKR_400Regular",
                    "fontSize": 14,
                    "height": 80.03999999999999,
                    "includeFontPadding": false,
                    "textAlign": "center",
                    "textAlignVertical": "center",
                    "width": 375,
                  },
                  undefined,
                ],
              ]
            }
          >
            동네 1
          </Text>
          <View />
        </View>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "height": 133.4,
        "justifyContent": "center",
      }
    }
  >
    <View
      adjustFrame={[Function]}
      animated={true}
      defaultIndex={-1}
      defaultValue="동네 2"
      disabled={false}
      dropdownStyle={
        Object {
          "height": 0,
          "width": 375,
        }
      }
      dropdownTextStyle={
        Object {
          "color": "black",
          "fontFamily": "NotoSansKR_300Light",
          "fontSize": 14,
          "height": 24,
          "includeFontPadding": false,
          "marginVertical": 5,
          "paddingVertical": 0,
          "textAlign": "center",
          "textAlignVertical": "center",
        }
      }
      isFullWidth={false}
      keyboardShouldPersistTaps="never"
      multipleSelect={false}
      onSelect={[Function]}
      options={Array []}
      renderButtonComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "render": [Function],
        }
      }
      renderRightComponent={[Function]}
      renderRowComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "render": [Function],
        }
      }
      scrollEnabled={true}
      showsVerticalScrollIndicator={true}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#FFFFFF",
          "borderRadius": 5,
          "height": 80.03999999999999,
          "justifyContent": "center",
          "width": 375,
        }
      }
      testID="modal-drop-down"
      textStyle={
        Object {
          "color": undefined,
          "fontFamily": "NotoSansKR_400Regular",
          "fontSize": 14,
          "height": 80.03999999999999,
          "includeFontPadding": false,
          "textAlign": "center",
          "textAlignVertical": "center",
          "width": 375,
        }
      }
    >
      <View
        accessible={true}
        focusable={true}
        forwardedRef={[Function]}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
            }
          }
        >
          <Text
            numberOfLines={1}
            style={
              Array [
                Object {
                  "fontSize": 12,
                },
                Array [
                  Object {
                    "color": undefined,
                    "fontFamily": "NotoSansKR_400Regular",
                    "fontSize": 14,
                    "height": 80.03999999999999,
                    "includeFontPadding": false,
                    "textAlign": "center",
                    "textAlignVertical": "center",
                    "width": 375,
                  },
                  undefined,
                ],
              ]
            }
          >
            동네 2
          </Text>
          <View />
        </View>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "height": 133.4,
        "justifyContent": "center",
      }
    }
  >
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#283593",
          "borderRadius": 10,
          "height": 80.03999999999999,
          "justifyContent": "center",
          "marginBottom": undefined,
          "marginLeft": undefined,
          "marginRight": undefined,
          "marginTop": undefined,
          "opacity": 1,
          "width": 375,
        }
      }
      testID="custom-button"
    >
      <Text
        style={
          Object {
            "color": "#FFFFFF",
            "fontFamily": "NotoSansKR_400Regular",
            "fontSize": 18,
            "height": "100%",
            "includeFontPadding": false,
            "textAlign": "center",
            "textAlignVertical": "center",
            "width": "100%",
          }
        }
      >
        등록하기
      </Text>
    </View>
  </View>
</RNCSafeAreaView>
`;
